xclim/indices/fire/_ffdi.py

```diff
@@ -0,0 +1,393 @@
+        # Limit kbdi to between 0 and 200 mm
+        if kbdi0 < 0.0:
+            kbdi0 = 0.0
+
+        if kbdi0 > 203.2:
+            kbdi0 = 203.2
+
+        rr = rr - r
+        kbdi[d] = kbdi0
+
+
+@guvectorize(
+    [
+        (
+            float64[:],
+            float64[:],
+            int64,
+            float64[:],
+        )
+    ],
+    "(n),(n),()->(n)",
+)
+def _griffiths_drought_factor(p, smd, lim, df):  # pragma: no cover
+    """Compute the Griffiths drought factor.
+
+    Parameters
+    ----------
+    p : array_like
+        Total rainfall over previous 24 hours [mm].
+    smd : array_like
+        Soil moisture deficit (e.g. KBDI).
+    lim : integer
+        How to limit the drought factor.
+        If 0, use equation (14) in :cite:t:`ffdi-finkele_2006`.
+        If 1, use equation (13) in :cite:t:`ffdi-finkele_2006`.
+
+    Returns
+    -------
+    df : array_like
+        The limited Griffiths drought factor
+    """
+    wl = 20  # 20-day window length
+
+    for d in range(wl - 1, len(p)):
+        pw = p[d - wl + 1 : d + 1]
+
+        # Calculate the x-function from significant rainfall
+        # events
+        conseq = 0
+        pmax = 0.0
+        P = 0.0
+        x = 1.0
+        for iw in range(wl):
+            event = pw[iw] > 2.0
+            event_end = ~event & (conseq != 0)
+            final_event = event & (iw == (wl - 1))
+
+            if event:
+                conseq = conseq + 1
+                P = P + pw[iw]
+                if pw[iw] >= pmax:
+                    N = wl - iw
+                    pmax = pw[iw]
+
+            if event_end | final_event:
+                # N = 0 defines a rainfall event since 9am today,
+                # so doesn't apply here, where p is the rainfall
+                # over previous 24 hours.
+                x_ = N**1.3 / (N**1.3 + P - 2.0)
+                x = min(x_, x)
+
+                conseq = 0
+                P = 0.0
+                pmax = 0.0
+
+        if lim == 0:
+            if smd[d] < 20:
+                xlim = 1 / (1 + 0.1135 * smd[d])
+            else:
+                xlim = 75 / (270.525 - 1.267 * smd[d])
+            if x > xlim:
+                x = xlim
+
+        dfw = (
+            10.5
+            * (1 - np.exp(-(smd[d] + 30) / 40))
+            * (41 * x**2 + x)
+            / (40 * x**2 + x + 1)
+        )
+
+        if lim == 1:
+            if smd[d] < 25.0:
+                dflim = 6.0
+            elif (smd[d] >= 25.0) & (smd[d] < 42.0):
+                dflim = 7.0
+            elif (smd[d] >= 42.0) & (smd[d] < 65.0):
+                dflim = 8.0
+            elif (smd[d] >= 65.0) & (smd[d] < 100.0):
+                dflim = 9.0
+            else:
+                dflim = 10.0
+            if dfw > dflim:
+                dfw = dflim
+
+        if dfw > 10.0:
+            dfw = 10.0
+
+        df[d] = dfw
+
+
+# SECTION 2 - Public methods and indices
+
+
+def _keetch_byram_drought_index_pass(pr, tasmax, pr_annual, kbdi0):
```

Comment:
I would declare this function inside the main one (`keetch_byram_drought_index`) to "remove" it from the module and avoid any confusion with the vectorized one. Same thing with the drought factor one.